packages/grafana-data/src/transformations/transformers/calculateField.test.ts

```diff
@@ -728,6 +728,39 @@ describe('calculateField transformer w/ timeseries', () => {
     });
   });
 
+  it('calculates fixed, trailing moving average with missing values', async () => {
+    const cfg = {
+      id: DataTransformerID.calculateField,
+      options: {
+        mode: CalculateFieldMode.WindowFunctions,
+        window: {
+          windowAlignment: WindowAlignment.Trailing,
+          field: 'x',
+          windowSize: 2,
+          windowSizeMode: WindowSizeMode.Fixed,
+          reducer: ReducerID.mean,
+        },
+      },
+    };
+
+    const series = toDataFrame({
+      fields: [{ name: 'x', type: FieldType.number, values: [1, undefined, 3, 4, 5] }],
+    });
+
+    await expect(transformDataFrame([cfg], [series])).toEmitValuesWith((received) => {
+      const data = received[0][0];
+
+      //console.log(data.fields);
+
+      expect(data.fields.length).toEqual(2);
```

Comment:
@leeoniya I feel like this test is right but the results are wrong. Should it be 1, 0.5, 1.5, 3.5, 4.5 ?

Reply:
yeah, agreed. these results look like a rolling sum, not mean.

Reply:
Good news: I fixed it by ensuring that the current value the window lands on also evaluates to 0 if it is null. I then get what I'd expect.

Bad news: this did end up changing the results of another test. I do agree with the change, but I could see it being problematic if the previous results were evaluated to be correct at the time. Not sure how to proceed to be honest.